coil-gif/src/main/java/coil/decode/ImageDecoderDecoder.kt

```diff
@@ -108,6 +112,22 @@ class ImageDecoderDecoder : Decoder {
             val drawable = if (baseDrawable is AnimatedImageDrawable) {
                 baseDrawable.repeatCount = options.parameters.repeatCount() ?: AnimatedImageDrawable.REPEAT_INFINITE
 
+                // Set the start and end animation callbacks if any one is supplied through the request.
+                if (options.parameters.animationStartCallback() != null ||
```

Comment:
Nit: I would set `options.parameters.animationStartCallback()` to a local variable here (and in `GifDecoder`) so Kotlin infers that it's not-null inside the callback.

Reply:
Auto infer won't work here as there is `||` condition.